python/ray/autoscaler/_private/node_launcher.py

```diff
@@ -123,31 +93,68 @@ def _launch_node(
         if node_type:
             node_tags[TAG_RAY_USER_NODE_TYPE] = node_type
             node_config.update(launch_config)
-        launch_start_time = time.time()
+
+        node_launch_start_time = time.time()
+
+        error_msg = None
+        full_exception = None
         try:
             self.provider.create_node_with_resources(
                 node_config, node_tags, count, resources
             )
         except NodeLaunchException as node_launch_exception:
             self.node_provider_availability_tracker.update_node_availability(
-                node_type, launch_start_time, node_launch_exception
+                node_type, int(node_launch_start_time), node_launch_exception
             )
-            # Do some special handling if we have a structured error.
-            self.log(
-                f"Failed to launch {node_type}: "
-                f"({node_launch_exception.category}):"
+
+            if node_launch_exception.src_exc_info is not None:
+                full_exception = "\n".join(
+                    traceback.format_exception(*node_launch_exception.src_exc_info)
+                )
+
+            error_msg = (
+                f"Failed to launch {{}} node(s) of type {node_type}. "
```

Comment:
Not an extremely critical distinction, just observing:
it may be that you've failed to launch only 1 out of `count` nodes in this situation.